[ATO-663] Port solution to compress custom action request to main (#11998)
RasaHQ/rasa: docs/docs/action-server/actions.mdx

The action will set the `temperature` to the maximum temperature of the weather forecast.
The message returned will differ according to the channel they are using. 

:::info Compressed body in HTTP requests

> Rasa has capability to send compressed body in HTTP requests for custom actions.
> By default, this option is off to keep backward compatibility with older versions of
> custom action servers which did not receive compressed body in HTTP request for custom actions.
> To enable this option set environment variable COMPRESS_ACTION_SERVER_REQUEST to True.
> 
> Rasa SDK 3.2.2, 3.3.1, 3.4.1 and onwards,

Reviewer: ```suggestion
Rasa has the capability to compress the HTTP request body for custom actions.
By default, this option is off to keep backward compatibility with older versions of
custom action servers which did not receive the compressed body in the HTTP request for custom actions.
To enable this option, set environment variable COMPRESS_ACTION_SERVER_REQUEST to `True`.

Rasa SDK versions `3.2.2`, `3.3.1`, `3.4.1` and upwards,
```

> support both compressed and non-compressed body in HTTP request for

Reviewer: ```suggestion
support both compressed and non-compressed body in the HTTP request for
```

running custom action server. There is no additional setup required.

> 

Reviewer: Missing closing info note
```suggestion
:::
```
